### ISSUE TITLE
The problem about aug_loss

### ISSUE THREAD

**Issue by yangpanquan** [ShiQiu0419/BAAF-Net#9]
Hi, it is a good job.I am interested in it! But I am confused with  aug_loss.
The code is followed:

 aug_loss_weights = tf.constant([0.1, 0.1, 0.3, 0.5, 0.5])
 aug_loss = 0           # new_xyz_list=（B,N,16,3）...xyz_list=（B,N,3)...
 for i in range(self.config.num_layers):
          centroids = tf.reduce_mean(self.new_xyz[i], axis=2)  # (B,N,3)
          relative_dis = tf.sqrt(tf.reduce_sum(tf.square(centroids-self.xyz[i]), axis=-1) + 1e-12) # （B,N,1）
           aug_loss = aug_loss + aug_loss_weights[i] * tf.reduce_mean(tf.reduce_mean(relative_dis, axis=-1), axis=-1)  #weight*B

Q1: why is the aug_loss_weight assigned like this？
Q2: the output of code" tf.reduce_mean(tf.reduce_mean(relative_dis, axis=-1), axis=-1)" is B?  Is my remarks correct? I hope to get your reply.

**Comment by ShiQiu0419** [comment]
Hi @yangpanquan ,

For your Q1: 
They are empirical weights based on my coarse tests, where the sum of augmentation losses need to be relatively smaller than the main cross-entropy loss. On the other hand, as the U-Net goes deeper, the neighborhood is sparse, thus we need higher weight to force the neighbors approaching the centroids.

For your Q2: 
No, it is a number.
Say in: https://github.com/ShiQiu0419/BAAF-Net/blob/663d1681d4d05ad3caaacd98e6dedfdc9caa4930/BAAFNet.py#L98 we have ```self.new_xyz[i]``` in shape [B,N,k,3], then ```centroids``` is in [B,N,3]. 
Further, in: https://github.com/ShiQiu0419/BAAF-Net/blob/663d1681d4d05ad3caaacd98e6dedfdc9caa4930/BAAFNet.py#L99 we get ```relative_dis``` in [B,N]. 
Finally the output of ```tf.reduce_mean(tf.reduce_mean(relative_dis, axis=-1), axis=-1)``` becomes a real number indicating an augmentation loss.

**Comment by yangpanquan** [comment]
Thank you for your detail reply. So, the output of "tf.reduce_mean(tf.reduce_mean(relative_dis, axis=-1), axis=-1)" is a mean 
 relative distance to centroids  in the  total relative distance(BxN).is it all right?Besides , I have another problem.What are your considerations for using aug_loss.After seeing the part of augmentation loss in the paper, I can not understand it.

**Comment by ShiQiu0419** [comment]
Yes you are right.

In the bilateral augmention part, we try to learn the offsets, which intend to make a reasonable "virtual" neighborhood by pulling all neighbors closer to the centroid. However, the learning process of offsets is pretty tricky. Thus, we need something physically meaningful to guide the learning of offsets.

**Comment by yangpanquan** [comment]
Ok,  I get it .Thank for you detail reply again! 
Have a good day!
